src/werkzeug/datastructures.py

```diff
@@ -602,6 +602,9 @@ def popitemlist(self):
         except KeyError as e:
             raise exceptions.BadRequestKeyError(e.args[0])
 
+    def __iter__(self):
+        yield from dict.keys(self)
```

Comment:
`yield from super().keys()` would also fix this issue, and better allow cooperative inheritance. But there are several other places in this file that hardcode superclass references. Curious if anyone knows why, even though it’s not necessary to change at this time.

Reply:
Actually, please take a look at #2043 for a more minimal fix.